enhancements/kube-apiserver/stability-point-to-point-network-check.md

```diff
@@ -0,0 +1,327 @@
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+
+#### Story 1
+
+#### Story 2
+
+### Implementation Details/Notes/Constraints [optional]
+
+What are the caveats to the implementation? What are some important details that
+didn't come across above. Go in to as much detail as necessary here. This might
+be a good place to talk about core concepts and how they relate.
+
+#### RelatedEndpoint Custom Resource
+
+```go
+package v1alpha1
+
+import (
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// ConditionCheck
+type ConditionCheck struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata"`
+
+	Spec   ConditionCheckSpec   `json:"spec"`
+	Status ConditionCheckStatus `json:"status"`
+}
+
+type ConditionCheckSpec struct {
+	// SourcePod names the pod from which the condition will be checked
+	SourcePod string `json:"source_pod"`
+	// Endpoint being checked
+	Endpoint string `json:"endpoint"`
+	// Type of check performed on the endpoint
+	Type ConditionCheckType `json:"type"`
+}
+
+type ConditionCheckStatus struct {
+	// Log of check events
+	Log []LogEntry `json:"log"`
+	// Conditions summarize the status of the check
+	Conditions []ConditionCheckCondition `json:"conditions"`
+}
+
+// LogEntry records events
+type LogEntry struct {
+	Time     metav1.Time    `json:"time,omitempty"`
+	Status   LogEntryStatus `json:"status"`
+	Message  string         `json:"message"`
+	Duration time.Duration  `json:"duration"`
+}
+
+type ConditionCheckType string
+
+type LogEntryStatus string
+
+type ConditionCheckConditionType string
+
+type ConditionCheckCondition struct {
+	Type               ConditionCheckConditionType `json:"type"`
+	Status             ConditionStatus             `json:"status"`
+	LastTransitionTime metav1.Time                 `json:"lastTransitionTime,omitempty"`
+	Reason             string                      `json:"reason,omitempty"`
+	Message            string                      `json:"message,omitempty"`
+}
+
+const (
+	Reachable          ConditionCheckConditionType = "Reachable"
+	Pass               LogEntryStatus              = "Pass"
+	Fail               LogEntryStatus              = "Fail"
+	TCPEndpointConnect ConditionCheckType          = "TCPEndpointConnect"
+)
+
+type ConditionStatus string
+
+const (
+	ConditionTrue    ConditionStatus = "True"
+	ConditionFalse   ConditionStatus = "False"
+	ConditionUnknown ConditionStatus = "Unknown"
+)
+
+```
+
+
+#### Point-to-point Network Check Tool
+
+Runs in a container in the interested pod. Executes check and updates `ConditionCheck` resources for the pod.
+
+Usage: 
+
+```
+p2pnc check tcp-endpoint [list of endpoints]
+
+``` 
+
+Example:
+
+```
+p2pnc check tcp-endpoint \
+    10.0.100.1:2379 \
+    10.0.200.3:2379 \
+    10.0.300.5:2379
+```
+
+Running the above example command would create and then update a `ConditionC`
+
+```yaml
+Kind: ConditionCheck
+Version: network.openshift.io/v1alpha1
```

Comment:
@danwinship @knobunc willing to share your namespace? The goal of this is a point-to-point network check.